waspc/data/Generator/templates/react-app/src/queryClient.js

```diff
@@ -1,3 +1,16 @@
 import { QueryClient } from 'react-query'
 
-export const queryClient = new QueryClient()
+let resolveQueryClientInitialized, isQueryClientSetUp
+export const queryClientInitialized = new Promise(resolve => {
+  resolveQueryClientInitialized = resolve
+});
+
+export function setupQueryClient(config = {}) {
+  if (isQueryClientSetUp) {
```

Comment:
Notice the subtle difference between `setup` and `setUp`. If only "set" was a regular verb, a lot of code would gain in readability.